### PR TITLE
[IMP] hr_holidays: links accrual to company

### PR DESCRIPTION
**Pull request by Bertrand2**
This commit adds a company field to accruals.

TaskID 2817577

### CODE REVIEW

File: addons/hr_holidays/security/hr_holidays_security.xml [hunk 1]
```diff
@@ -245,6 +245,12 @@
         <field name="domain_force">['|', ('company_id', '=', False), ('company_id', 'in', company_ids)]</field>
     </record>
 
+    <record id="hr_leave_accrual_plan_rule_multi_company" model="ir.rule">
+        <field name="name">Accrual plan multi company rule</field>
+        <field name="model_id" ref="model_hr_leave_accrual_plan"/>
+        <field name="domain_force">['|', ('company_id', '=', False), ('company_id', 'in', company_ids)]</field>
```
Comment (kba-odoo) [inline addons/hr_holidays/security/hr_holidays_security.xml:251]: As the field is required, the company_id = False can be removed

```suggestion
        <field name="domain_force">[('company_id', 'in', company_ids)]</field>
```

**Comment by kba-odoo** [comment]
robodoo r+

**Comment by kba-odoo** [comment]
robodoo r+

File: addons/hr_holidays/models/hr_leave_accrual_plan.py [hunk 2]
```diff
@@ -54,6 +59,14 @@ def _compute_employee_count(self):
         for plan in self:
             plan.employees_count = allocations_dict.get(plan.id, 0)
 
+    @api.depends('time_off_type_id')
```
Comment (sofiagvaladze) [inline addons/hr_holidays/models/hr_leave_accrual_plan.py:62]:
```suggestion
    @api.depends('time_off_type_id.company_id')
```

File: addons/hr_holidays/views/hr_leave_accrual_views.xml [hunk 3]
```diff
@@ -87,14 +87,17 @@
                             <field name="name"/>
                         </group>
                         <group>
+                            <field name="company_id" invisible="1"/>
```
Comment (sofiagvaladze) [inline addons/hr_holidays/views/hr_leave_accrual_views.xml:90]: No need to put invisible field, because we already have the field in the view :innocent:

Comment (Bertrand2) [reply addons/hr_holidays/views/hr_leave_accrual_views.xml:90]: But the present company_id field is not available unless you're in the base.group_multi_company group, so I think it still is required

Comment (sofiagvaladze) [reply addons/hr_holidays/views/hr_leave_accrual_views.xml:90]: You are right :innocent:

File: addons/hr_holidays/models/hr_leave_accrual_plan.py [hunk 4]
```diff
@@ -10,11 +10,15 @@ class AccrualPlan(models.Model):
...
     time_off_type_id = fields.Many2one('hr.leave.type', string="Time Off Type",
+        check_company=True,
         help="""Specify if this accrual plan can only be used with this Time Off Type.
                 Leave empty if this accrual plan can be used with any Time Off Type.""")
     employees_count = fields.Integer("Employees", compute='_compute_employee_count')
     level_ids = fields.One2many('hr.leave.accrual.level', 'accrual_plan_id', copy=True)
     allocation_ids = fields.One2many('hr.leave.allocation', 'accrual_plan_id')
+    company_id = fields.Many2one('res.company', string='Company',
+        compute="_compute_company_id", store="True",
+        readonly=False, default=lambda self: self.env.company)
```
Comment (kba-odoo) [inline addons/hr_holidays/models/hr_leave_accrual_plan.py:21]:
```suggestion
        compute="_compute_company_id", store=True,
        readonly=False)
```

Comment (kba-odoo) [reply addons/hr_holidays/models/hr_leave_accrual_plan.py:21]: Don't set a default value on a compute, instead you can set the default value you want in the compute itself

**Review by kbapt — CHANGES_REQUESTED** [review]

**Comment by kba-odoo** [comment]
robodoo r+

**Pull request closed without merging**